elastic/kibana
# [Uptime] hydrate service monitors with port data [#121125]

## Summary

This PR hydrates saved objects with `url.port` as specified in https://github.com/elastic/kibana/issues/121125.

There are, however, a few minor issues with the previous implementation which are still outstanding. They are described [here](https://github.com/elastic/kibana/issues/121125#issuecomment-1062829386).

## How to test this PR

1. Create a monitor (using the monitor management UI) which takes a significant amount of time to run, just so that you have time to check its `savedObject` before the task runner executes and updates it. One example could be:
    ```js
    step("test", async () => {
        await page.goto("https://bloomberg.com");
        await page.waitForTimeout(120000);
    });
    ```
2. Go to the saved objects page in the stack management page.
3. Filter for `synthetics-monitor` using the `type` filter to find your monitor.
4. Using the three dots on the right, see the saved object's attributes and confirm it doesn't have a `url.port` field.
5. Wait for approximately 5-6 minutes until the monitor has run and the `taskManager` has had the chance to hydrate the object.
6. Refresh the page and confirm that the object has now been updated with a `url.port` and `urls`.

⚠️ Please do the same for a TCP and HTTP monitors.


### Checklist

Delete any items that are not applicable to this PR.



### For maintainers


## Release note

Enrich documents generated by the synthetics service with `port` information.
